IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m

```diff
@@ -504,38 +504,14 @@ - (MSIDLegacyRefreshToken *)getRefreshTokenForAccountImpl:(MSIDAccountIdentifier
 
-    MSIDLegacyRefreshToken *resultToken = (MSIDLegacyRefreshToken *)[self getTokenByLegacyUserId:account.legacyAccountId
-                                                                                            type:MSIDRefreshTokenType
-                                                                                       authority:configuration.authority
-                                                                                   lookupAliases:aliases
-                                                                                        clientId:clientId
-                                                                                        resource:nil
-                                                                                         context:context
-                                                                                           error:error];
-
-    // If no legacy user ID available, or no token found by legacy user ID, try to look by unique user ID
-    if (!resultToken
-        && ![NSString msidIsStringNilOrBlank:account.homeAccountId])
-    {
-        MSID_LOG_VERBOSE(context, @"(Legacy accessor) Finding refresh token with new user ID, clientId %@, authority %@", clientId, aliases);
-        MSID_LOG_VERBOSE_PII(context, @"(Legacy accessor) Finding refresh token with new user ID %@, clientId %@, authority %@", account.homeAccountId, clientId, aliases);
-
-        resultToken = (MSIDLegacyRefreshToken *) [self getTokenByHomeAccountId:account.homeAccountId
-                                                                     tokenType:MSIDRefreshTokenType
-                                                                     authority:configuration.authority
-                                                                 lookupAliases:aliases
-                                                                      clientId:clientId
-                                                                      resource:nil
-                                                                       context:context
-                                                                         error:error];
-
-        if (resultToken && error)
-        {
-            *error = nil;
-        }
-    }
-
-    return resultToken;
+    return (MSIDLegacyRefreshToken *)[self getTokenByLegacyUserId:account.legacyAccountId
+                                                             type:MSIDRefreshTokenType
+                                                        authority:configuration.authority
+                                                    lookupAliases:aliases
+                                                         clientId:clientId
+                                                         resource:nil
+                                                          context:context
+                                                            error:error];
```

Comment:
Shall we add some comments about this logic change before the function? As it is hard to know by function name that it only tries the `legacyAccountId` in the `account` parameter :)

Reply:
sure, will rename the method :)